core/templates/services/math-interactions.service.ts

```diff
@@ -81,11 +82,23 @@ export class MathInteractionsService {
         'Your answer has two symbols next to each other: "' + symbol1 +
         '" and "' + symbol2 + '".');
     }
+    if (
+      errorMessage === 'Cannot read property \'parent\' of undefined.') {
+      let invalidFunctionName = '';
+      for (let functionName of this.mathFunctionNames) {
+        if (expressionString.includes(functionName + '()')) {
+          invalidFunctionName = functionName;
+          break;
+        }
+      }
```

Comment:
Can there be more invalid function names?

Reply:
Oh -- the function name itself isn't invalid. I named it as invalid since it's being called in an invalid manner (without any parameters passed in). I've renamed this to `emptyFunctionName` to avoid this confusion.

Reply:
Yeah I know, what I mean is when there are multiple empty functions (but the naming is now probably a bit better too).

Reply:
I see. Yes, there could be multiple invalid functions in the expressions like `sin()tan()` and in that case, the error message will show one of them until all of them have been resolved. I've also added a test case for this.

Reply:
Hmm, I think it might be better to show all the invalid at once, but if that requires complicated changes then it should be fine.

Reply:
Done. Now, all empty function names will be displayed.